### ISSUE TITLE
Output graphs are so jumbled.? 

### ISSUE THREAD

**Issue by DRAhmadFaraz** [NVlabs/geomapnet#13]
Sir I run your code to Evaluate the pretrained Mapnet++ on Heads, Offices of 7 scenes and I get such jumbled output as compared to The output u show me in your research paper.

![Screenshot from 2019-03-11 15-43-46](https://user-images.githubusercontent.com/38114191/54187772-3078e500-44a6-11e9-83d7-0e1079824a71.png)
![Screenshot from 2019-03-11 15-58-25](https://user-images.githubusercontent.com/38114191/54187777-31117b80-44a6-11e9-84bb-89ec1233b286.png)
![Screenshot from 2019-03-11 16-16-17](https://user-images.githubusercontent.com/38114191/54187778-31aa1200-44a6-11e9-9015-3818da41b483.png)
![Screenshot from 2019-03-11 16-20-13](https://user-images.githubusercontent.com/38114191/54187780-3242a880-44a6-11e9-972e-2d1685141bae.png)

The output should not be just like in your research paper as below.?

![Screenshot from 2019-03-12 09-07-32](https://user-images.githubusercontent.com/38114191/54187859-57cfb200-44a6-11e9-830b-0475f2b5c06d.png)

**Comment by fast785** [comment]
I also faced the same issue,, @samarth-robo, Kindly guide me.

**Comment by samarth-robo** [comment]
Hi @DRAhmadFaraz, I see from the first screenshot that you are using the model trained on the  `heads` scene and applying it to the `office` scene. We have separate models for each scene.
So can you try the following command and post the output?

```
$ python eval.py --dataset 7Scenes --scene heads --model mapnet++ \
--weights logs/7Scenes_heads_mapnet++_mapnet++_7Scenes/epoch_005.pth.tar \
--config_file configs/pgo_inference_7Scenes.ini --val --pose_graph
```

I can verify that it gives the expected numbers on my computer, and I'll check the visualization when I have visual access to my computer tomorrow.

**Comment by samarth-robo** [comment]
My output matches the paper:
```
(mapnet_release) $ python eval.py --dataset 7Scenes --scene heads --model mapnet++ --weights logs/7Scenes_heads_mapnet++_mapnet++_7Scenes/epoch_005.pth.tar --val --config configs/pgo_inference_7Scenes.ini --pose_graph
Loaded weights from logs/7Scenes_heads_mapnet++_mapnet++_7Scenes/epoch_005.pth.tar
Running mapnet++ on VAL data
Image 0 / 988
Image 200 / 988
Image 400 / 988
Image 600 / 988
Image 800 / 988
Error in translation: median 0.12 m,  mean 0.13 m
Error in rotation: median 8.46 degrees, mean 9.15 degree
```
![heads](https://user-images.githubusercontent.com/2848070/54724245-b220e080-4b40-11e9-86f3-f5545de455bc.png)

Please make sure you have downloaded the 7Scenes data properly, and followed the instructions in the repository's readme. I'll close this issue now, feel free to re-open it if the problem persists.

**Comment by m5823779** [comment]
Hello, I have downloaded your source code, then downloaded 7scense dataset in "/data/deepslam_data/7Scenes" and downloaded trained models from google drive.
But my result still jumbled, although I follow your step.
By the way, I didn't use Conda environment
Following is my result,
![螢幕快照 2019-03-27 16:41:04](https://user-images.githubusercontent.com/34086087/55061860-beacb800-50af-11e9-9ad1-cd2a09de16e3.png)

thanks~

**Comment by DRAhmadFaraz** [comment]
@samarth-robo  

Sir after typing this command exactly as it is 

$ python eval.py --dataset 7Scenes --scene heads --model mapnet++ \
--weights logs/7Scenes_heads_mapnet++_mapnet++_7Scenes/epoch_005.pth.tar \
--config_file configs/pgo_inference_7Scenes.ini --val --pose_graph

I get the following jumbled results still.


![Screenshot from 2019-03-27 08-59-17](https://user-images.githubusercontent.com/38114191/55062883-b2ecd180-506e-11e9-94cd-3abd667b4d2b.png)

**Comment by DRAhmadFaraz** [comment]
@samarth-robo Sir, I am waiting for your response.

**Comment by samarth-robo** [comment]
@DRAhmadFaraz @m5823779 I also re-downloaded the `heads` dataset and verified that the model gives the correct results. I'm not sure what's causing this issue for you.

To rule out the possibility of library version problems, can you try again after installing the conda environment as mentioned in the readme?

**Comment by DRAhmadFaraz** [comment]
Yes Sir, I have tried again using anaconda environment (instead of mini-conda) .

I downloaded and stored the dataset in directory "data/deepslam_data".
Then I created the following symlinks

cd data/deepslam_data && ln -s 7SCENES_DIR 7Scenes && ln -s ROBOTCAR_DIR RobotCar_download

Then I do

python eval.py --dataset 7Scenes --scene heads --model mapnet++ 
--weights logs/7Scenes_heads_mapnet++_mapnet++_7Scenes/epoch_005.pth.tar 
--config_file configs/pgo_inference_7Scenes.ini --val --pose_graph

the evaluation start and gives me jumbled outputs.

**Comment by m5823779** [comment]
@DRAhmadFaraz @samarth-robo Hello~ I think I have solved this problem.
I just train the network again by myself, and following is my testing result.

![image](https://user-images.githubusercontent.com/34086087/55308634-e6c06080-548d-11e9-9397-44dab3dad310.png)

thanks for your help

**Comment by samarth-robo** [comment]
@DRAhmadFaraz @m5823779 I reinstalled the conda env and re-downloaded the dataset on a fresh computer, and it still gives me the correct results.
I had to downgrade pytorch to version 0.4.1 though. Are you using version 1.0? That might be the problem. I've pushed a change to the environment.yml file that will force the pytorch version to be 0.4.1.

**Comment by ez4lionky** [comment]
> @DRAhmadFaraz @m5823779 I reinstalled the conda env and re-downloaded the dataset on a fresh computer, and it still gives me the correct results.
> I had to downgrade pytorch to version 0.4.1 though. Are you using version 1.0? That might be the problem. I've pushed a change to the environment.yml file that will force the pytorch version to be 0.4.1.

Actually, the problem can be solved by changing the code of line 69 in posenet.py to:
x = F.dropout(x, p=self.droprate, training=self.training).
In fact, the dropout layer is invalid when using torch.nn.functional() but not setting the training parameter.
The difference between pytorch 1.0 and 0.4.0 is, in 1.0, the default value of the training parameter is set to True, instead of False in 0.4.0.
